_health-care/_js/_form.js

```diff
@@ -18,17 +18,60 @@ var HealthApp = (function() {
     var dateString = "";
 
     if (date.getMonth() < 10) {
-      dateString += "0"
+      dateString += "0";
     }
     dateString += (date.getMonth() + 1) + "/";
 
     if (date.getDate() < 10) {
-      dateString += "0"
+      dateString += "0";
     }
     dateString += date.getDate() + "/";
 
     return dateString + date.getFullYear();
   }
+  function extractMonetaryValueTransform(formData, arg) {
+    return Number(extractTransform(formData, arg)).toFixed(2).toString();
+  }
+  function validatePresence(input) {
+    var valid = true;
+
+    if (input.value === '') {
+      valid = false;
+    }
+
+    return valid;
+  }
+  function validateMonetaryValue(input) {
+    var valid = true;
+    var amount = input.value;
+
+    if (/[$a-zA-Z,]+/.test(amount)) {
```

Comment:
In general, prefer whitelists to blacklists. For this code, I think you'd still return "valid" for random symbols like `%#` or most unicode characters.

So, I think, /\d+\.?\d*/.  There should be a unittest on this function testing the various input patterns. At least cases for stuff like `10`, and `11.`, and `12.3` and then failure for `12.3.4`.  Don't know if we should allow `.1` etc....

Also, this might be helpful. The validation logic used in the original XFA PDF form is here:
```
function validAmount(AmountIn) {
        if (AmountIn.isNull==true)
                return(true);
        var strAmount = String(AmountIn.rawValue)
        var sTemp="";
        for (var i=0; i&lt;strAmount.length; i++) {
                var sChar =  strAmount.substr(i, 1);
            if (isNumeric(sChar) || (sChar==".")) sTemp = sTemp + sChar;
        }       
        if ((sTemp==strAmount) &amp;&amp; (AmountIn.rawValue &lt;= 9999999.99)) {
                return(true)
        }       
        else {
                return(false);
        }                       
}
```
(ripped from https://raw.githubusercontent.com/department-of-veterans-affairs/1010ez-er/master/doc/xml-samples/anonymous-1010ez-obj27-template.xml).

I wouldn't quite replicate their code...it's doing some horridly slow string manipulations (substr per character means O(n^2) strings created per validation. yikes!), but it seems like they check that it's less than 9999999.99 so might be worth putting that in?